pandas/core/arrays/base.py

```diff
@@ -1232,6 +1236,105 @@ def _repr_2d(self) -> str:
         class_name = f"<{type(self).__name__}>"
         return f"{class_name}\n[\n{data}\n]\nShape: {self.shape}, dtype: {self.dtype}"
 
+    def _format_array(
+        self,
+        formatter: Callable | None,
+        *,
+        float_format: FloatFormatType,
```

Comment:
should we take this opportunity to make a  FormatOptionsType that has all of these defaults?